integration/gradient-project-setup.py

```diff
@@ -0,0 +1,325 @@
+gradient_apply_rec_notebook_path='/Users/' + w.current_user.me().user_name + '/apply-gradient-recommendation'
+secret_scope='gradient_' + w.current_user.me().user_name
+generate_clone=True
+add_apply=False
+
+#### UPDATE THESE VALUES ####
+
+db_workspace_url='*** Update with your value ***'
+job_id='*** Update with your value ***'
+sync_project_id='*** Update with your value ***'
+sync_access_key_id='*** Update with your value ***'
+sync_secret_access_key='*** Update with your value ***'
+db_token='*** Update with your value ***'
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Download Required Files
+# MAGIC  - Gradient Agent (saved to path specified in workspace_path_init)
+# MAGIC  - Gradient Notebook (saved to path specified in gradient_notebook_path)
+
+# COMMAND ----------
+
+import base64
+import requests
+
+url = "https://raw.githubusercontent.com/synccomputingcode/client_tools/main/integration/gradient-agent.sh"
+file_contents = requests.get(url).text
+
+w.workspace.import_(content=base64.b64encode((file_contents).encode()).decode(),
+                    format=workspace.ImportFormat.AUTO,
+                    overwrite=True,
+                    path=workspace_path_init)
+
+# COMMAND ----------
+
+url = "https://raw.githubusercontent.com/synccomputingcode/client_tools/main/integration/send-logs-gradient.py"
+file_contents = requests.get(url).text
+
+w.workspace.import_(content=base64.b64encode((file_contents).encode()).decode(),
+                    format=workspace.ImportFormat.SOURCE,
+                    overwrite=True,
+                    language=workspace.Language.PYTHON,
+                    path=gradient_notebook_path)
+
+# COMMAND ----------
+
+if add_apply:
+    url = "https://raw.githubusercontent.com/synccomputingcode/client_tools/main/integration/apply-gradient-recommendation.py"
+    file_contents = requests.get(url).text
+
+    w.workspace.import_(content=base64.b64encode((file_contents).encode()).decode(),
+                        format=workspace.ImportFormat.SOURCE,
+                        overwrite=True,
+                        language=workspace.Language.PYTHON,
+                        path=gradient_notebook_path)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Create New Secret Key Value Pairs for Credentials
+
+# COMMAND ----------
+
+scope_name = secret_scope
+scopes = w.secrets.list_scopes()
+if any(x for x in scopes if x.name == scope_name):
+    print('Using existing scope:' + scope_name)
+else:
+    w.secrets.create_scope(scope=scope_name)
+    print(scope_name + ' created')
+
+# Update the values for the secrets above if you are not using existing secrets
+w.secrets.put_secret(scope=scope_name, key='db_token', string_value=db_token)
+w.secrets.put_secret(scope=scope_name, key='sync_access_key_id', string_value=sync_access_key_id)
+w.secrets.put_secret(scope=scope_name, key='sync_secret_access_key', string_value=sync_secret_access_key)
```

Comment:
@syncpete do you want to separate workspace integration from project setup? Seems like we would try to readd secrets every time the customer wants to add a new job which only needs to happen once per workspace

Reply:
Actually if youre setting up a project for each job then it doesn't really matter.